src/iOS/Xamarin.iOS/Samples/Layers/CreateFeatureCollectionLayer/CreateFeatureCollectionLayer.cs

```diff
@@ -0,0 +1,177 @@
+{
+    [Register("CreateFeatureCollectionLayer")]
+    public class CreateFeatureCollectionLayer : UIViewController
+    {
+        // Reference to the MapView used in the app
+        private MapView _myMapView;
+
+        public CreateFeatureCollectionLayer()
+        {
+            Title = "Create a new feature collection layer";
+        }
+
+        public override void ViewDidLoad()
+        {
+            base.ViewDidLoad();
+
+            // Create the layout
+            CreateLayout();
+
+            // Initialize the app
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            try
+            {
+                // Create a new map with the oceans basemap and add it to the map view
+                var map = new Map(Basemap.CreateOceans());
+                _myMapView.Map = map;
+
+                // Call a function that will create a new feature collection layer and zoom to it
+                CreateNewFeatureCollection();
+            }
+            catch (Exception ex)
+            {
+                UIAlertView alert = new UIAlertView("Error", "Unable to create feature collection layer: " + ex.Message, null, "OK");
+                alert.Show();
+            }
+        }
+
+        private void CreateLayout()
+        {
+            // Define an offset from the top of the page (to account for the iOS status bar)
+            var yPageOffset = 60;
+
+            // Create a new MapView
+            _myMapView = new MapView();
+
+            // Setup the visual frame for the MapView
+            _myMapView.Frame = new CoreGraphics.CGRect(
+                0, yPageOffset + 40, View.Bounds.Width, View.Bounds.Height - yPageOffset - 40);            
+
+            // Add the MapView to the page
+            View.AddSubview(_myMapView);
+        }
+
+        private async void CreateNewFeatureCollection()
+        {
+            // Create the schema for a points table (one text field to contain a name attribute)
+            var pointFields = new List<Field>();
+            var placeField = new Field(FieldType.Text, "Place", "Place Name", 50);
+            pointFields.Add(placeField);
+
+            // Create the schema for a lines table (one text field to contain a name attribute)
+            var lineFields = new List<Field>();
+            var boundaryField = new Field(FieldType.Text, "Boundary", "Boundary Name", 50);
+            lineFields.Add(boundaryField);
+
+            // Create the schema for a polygon table (one text field to contain a name attribute)
+            var polyFields = new List<Field>();
+            var areaField = new Field(FieldType.Text, "AreaName", "Area Name", 50);
+            polyFields.Add(areaField);
+
+            // Instantiate FeatureCollectionTables with schema and geometry type
+            var pointsTable = new FeatureCollectionTable(pointFields, GeometryType.Point, SpatialReferences.Wgs84);
```

Comment:
Thanks, I made this change wherever I found it. At least the VB versions were correct :).